src/models/model.cpp

```diff
@@ -555,8 +555,14 @@ std::unique_ptr<OrtValue> Model::ExpandInputs(std::unique_ptr<OrtValue>& input,
 
 MultiModalProcessor::MultiModalProcessor(Config& config, const SessionInfo& session_info)
     : tokenizer_{std::make_shared<Tokenizer>(config)} {
-  if (!config.model.vision.filename.empty()) {
-    image_processor_ = std::make_shared<ImageProcessor>(config, session_info);
+  if (config.model.type == "phi3v") {
+    if (!config.model.vision.filename.empty()) {
+      image_processor_ = std::make_shared<ImageProcessor>(config, session_info);
+    } else {
+      throw std::runtime_error("ImageProcessor cannot be created. Vision model not found.");
```

Comment:
Is `MultiModalProcessor` designed to work without vision capability? If so, this should probably be just a warning.

Reply:
MultiModalProcessor is designed to support vision capabilities for now. Audio capabilities will be added in the near future.
What capabilities the MultiModalProcessor supports is model type dependent. For example, phi3-vision should support creating the image processor. The whisper model should create the audio processor. And maybe other model types will require other types of processors in the future.

Maybe I can remove the check for the filename being empty since it is not used here. But certainly, if the model is phi3v, the image processor should be created.